src/HotChocolate/MongoDb/src/Data/Paging/MongoOffsetPagingProvider.cs

```diff
@@ -0,0 +1,146 @@
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using HotChocolate.Internal;
+using HotChocolate.Data.MongoDb.Execution;
+using HotChocolate.Resolvers;
+using HotChocolate.Types;
+using HotChocolate.Types.Pagination;
+using MongoDB.Driver;
+
+namespace HotChocolate.Data.MongoDb.Paging
+{
+    public class MongoOffsetPagingProvider : OffsetPagingProvider
+    {
+        private static readonly MethodInfo _createHandler =
+            typeof(MongoOffsetPagingProvider).GetMethod(
+                nameof(CreateHandlerInternal),
+                BindingFlags.Static | BindingFlags.NonPublic)!;
+
+        public override bool CanHandle(IExtendedType source)
+        {
+            return true;
```

Comment:
Is it correct to always return true and don't check the source type?
I use the `UseOffsetPaging` attribute to add the paging, and I can't set the provider in the attribute. So I add in my Startup `services.AddTransient<OffsetPagingProvider, MongoOffsetPagingProvider>();` and I have error on paging if I have simple IQueryable to page.

Reply:
If you want to use both, IQueryable and Mongo you have to register both
```csharp
.AddFiltering()
.AddSorting()
.AddMongoDbFiltering("mongodb")
.AddMongoDbSorting("mongodb")
```

You now have two filtering configurations registered 
You can use them as the following

```csharp

[UseOffsetPaging]
[UseFiltering]
[UseSorting]
public IQueryable<Foo> GetFoos([Service] FooService svc) => foo.AsQueryable();

[UseMongoDbOffsetPaging]
[UseFiltering("mongo")]
[UseSorting("mongo")]
public IExecutable<Foo> GetFoos([Service] FooService svc) => foo.AsExecutable();
```

I just realized i forgot the attributes :innocent:  But you can add them by yourself to test (or just use .UseMongoDbPaging()):
https://github.com/ChilliCream/hotchocolate/blob/18216282c3e68365e83e70d7dd72e5c25f736af8/src/HotChocolate/MongoDb/src/Data/Paging/UseMongoDbOffsetPagingAttribute.cs

Also, I renamed UseMongoPaging to UseMongoDbPaging

Reply:
Thanks. I will try. 

But maybe you should return false in the CanHandle if the type is not supported to avoid to have exception.

Reply:
I also just aligned all remaining names to be consistent. Everything with Mongo is now named MongoDb.